case-lib/lib.sh

```diff
@@ -47,11 +47,15 @@ func_lib_setup_kernel_last_line()
 SOF_LOG_COLLECT=0
 func_lib_start_log_collect()
 {
-    local is_etrace=${1:-0} ldcFile
-    ldcFile=/etc/sof/sof-$(sof-dump-status.py -p).ldc || {
-        >&2 dlogw "sof-dump-status.py -p to query platform failed"
-        return
-    }
+    local is_etrace=${1:-0} ldcFile="$SOFLDC"
+    if [[ -z "$SOFLDC" ]]; then
+        # Default sof-*.ldc path used by CI is /etc/sof/sof-PLATFORM.ldc
+        ldcFile=/etc/sof/sof-$(sof-dump-status.py -p).ldc || {
+            >&2 dlogw "Failed to query platform with sof-dump-status.py"
+            return 1
+        }
+        [[ ! -e "$ldcFile" ]] && die "LDC file not found, check SOFLDC environment variable or put your sof-*.ldc to /etc/sof"
```

Comment:
Shouldn't this test be after and outside the `if` statement?

If `sof-dump-status.py -p` fails right now you `die`, on the other hand if the user misconfigures `$SOFLDC` then you let the `$loggerBin` fail later. This is inconsistent, I think both cases should fail the same way. I'm not sure which way though they should both fail though: how does `$loggerBin` fail when there is a typo in`$SOFLDC`? If there's already good error message (I think a good one was added recently) then we don't need this other assert here; "less is more"

Also, I'm not sure this sub-function should `die`, in fact the previous failure above does not `die` it just `returns 1`. A `return 1` lets the caller decide (with `set -e` or anything else), `die` gives the caller no choice.

Finally, I prefer this code style because there's no negation and it reads like English:
```shell
assert_something || return 1
```

Reply:
@marc-hb If there is a typo in $SOFLDC, sof-logger will exit immediately with error message; if there is no typo, sof-logger will runs during the whole test case, we have to manually kill it after test case finished. Currently, error message of sof-logger is redirected to /dev/null, because when we kill it, we get noise in the log like this
```
/home/ubuntu/sof-test/case-lib/hijack.sh: line 118: 120976 Killed 
/usr/bin/sudo env 'PATH=/home/ubuntu/sof-test/tools:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin' /usr/bin/sof-logger -t -l /etc/sof/sof-cml.ldcbak -o /home/ubuntu/sof-test/logs/check-playback/2020-09-09-13:52:57-6300/slogger.txt
```
according to you comment, let's do this:
- use `ldcFile` every where
- don't die, return 1 if `sof-dump-status.py` failed or user misconfigures `ldcFile`
- move [[ ! -e "$ldcFile" ]] out of the first -z test.

I test this on my Helios, this way makes it quite user friendly.

Reply:
> Currently, error message of sof-logger is redirected to /dev/null,

This is really, really bad because it hides _all_ errors for sof-logger. Test code should never redirect anything to /dev/null.

> because when we kill it, we get noise in the log like this

Test logs are not meant to be beautiful, they're meant to be useful and readable. Confusion if any can be avoided with a simple "Killing sof-logger" message.